### ISSUE TITLE
[bug] Tutorial multi-config layout on Linux does not work as expected

### ISSUE THREAD

**Issue by npuichigo** [conan-io/conan#13550]
### Environment details

* Operating System+version: Ubuntu 20.04
* Compiler+version: gcc 8
* Conan version: 2.0.2
* Python version: 3.10

### Steps to reproduce

https://docs.conan.io/2/tutorial/consuming_packages/the_flexibility_of_conanfile_py.html#use-the-layout-method

In tutorial, the build folder on Linux show be `f"build/{str(self.settings.build_type)}"` according to 
```
self.folders.build = "build" if multi else f"build/{str(self.settings.build_type)}"
```

but the compiled `compressor` still in `build` instead of `./build/Release`

**Comment by memsharded** [comment]
Hi @npuichigo 

Thanks for your report.
The ``build/Release`` folder is only expected for ``msvc`` compiler, not for ``gcc``. If you read some lines above:
```python
        if self.settings.get_safe("compiler") == "msvc":
            multi = True
        else:
            multi = False
```

This is because VisualStudio is "multi-config" environment (build/Debug, build/Release), while gcc is single-config environment (build)

**Comment by npuichigo** [comment]
So I think this line should be fixed.

![image](https://user-images.githubusercontent.com/11533479/228178299-e4159e58-91d1-42e3-a59b-dd91c1b1aa1d.png)

**Comment by memsharded** [comment]
Yes, that should be fixed in the docs, I am already doing some PR to update the docs cmake version first: https://github.com/conan-io/docs/pull/3132, and will be improving that part.

Transferring this ticket to the docs repo, thanks!
